kai/llm_interfacing/model_provider.py

```diff
@@ -57,6 +57,28 @@ def __init__(
             case "ChatOpenAI":
                 model_class = ChatOpenAI
 
+                # NOTE(JonahSussman): This is a hack to prevent `max_tokens`
+                # from getting converted to `max_completion_tokens`
+
+                @property  # type: ignore[misc]
+                def _default_params(self: ChatOpenAI) -> dict[str, Any]:
+                    return super(ChatOpenAI, self)._default_params
+
+                ChatOpenAI._default_params = _default_params  # type: ignore[method-assign]
+
+                def _get_request_payload(
```

Comment:
does the configureable fields still work with this? sorry i don't know the specifics of the api enough

Reply:
Yes, they should AFAIK